### DISCUSSION TITLE
How to configure Renovate to create onboarding PRs on repos that do not have a supported package file?

### DISCUSSION THREAD

**Discussion by shenke-adsk** [renovatebot/renovate#27942] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub Enterprise. Renovate version: 37.173.0

### Please tell us more about your question or problem

I observed a discrepancy between our hosted Renovate bot on GHE and the Renovate bot on github.com. Our app encountered a repository that does not contain a supported package file, and skipped processing. But the Renovate bot on github.com still went ahead and created the onboarding PR. 

How is it configured to continue the onboarding without finding any package files? Is that controlled by some env variable? From my perspective, the preferred behaviour would be

1. Default to skip if no package file found
2. If user really wants to, provide them the option to have the onboarding PR created.  

### Logs (if relevant)

<details><summary>Logs</summary>

Renovate bot on github.com:

> DEBUG: Found 0 package file(s)
> DEBUG: Need to create onboarding PR

Renovate bot hosted on our GHE instance:
> DEBUG: Found 0 package file(s)
> Repository has no package files - skipping

**Comment by rarkins** [comment, marked as answer]
In the hosted app, it sets [`onboardingNoDeps`](https://docs.renovatebot.com/self-hosted-configuration/#onboardingnodeps) to `true` if the account has "Selected repos" instead of "All repos". Because if they've selected individual repos then it's clear that they want Renovate to run on the repo.

If the user really wants in other cases, they need to push a renovate.json to the default branch.
